zerver/templatetags/app_filters.py

```diff
@@ -88,16 +89,34 @@ def render_markdown_path(markdown_file_path, context=None):
             integration_dir = markdown_file_path.split('/')[0]
         elif 'integrations' in markdown_file_path.split('/'):
             integration_dir = splitext(basename(markdown_file_path))[0]
+        elif 'zulip_bots' in markdown_file_path:
+            integration_dir = basename(os.path.dirname(markdown_file_path))
 
         integration = context['integrations_dict'][integration_dir]
 
         context['integration_name'] = integration.name
         context['integration_display_name'] = integration.display_name
-        context['recommended_stream_name'] = integration.stream_name
+        if hasattr(integration, 'stream_name'):
+            context['recommended_stream_name'] = integration.stream_name
         if hasattr(integration, 'url'):
             context['integration_url'] = integration.url[3:]
 
     jinja = engines['Jinja2']
+
+    # Jinja2's loader classes have an attribute Loader.searchpath
+    # that contains the list of directories that need to be searched
+    # for templates. These are usually defined in the settings.py
+    # of a Django project. However, in the case of Markdown files
+    # that are in the zulip_bots package, the directories that
+    # these templates may live in are arbitrary, depending on where the
+    # package is installed. Therefore, they can't be specified in
+    # settings.py. In that case, we should append the directory
+    # to be searched for a particular bot's documentation to
+    # Loader.searchpath to avoid a TemplateNotFound exception.
+    if 'zulip_bots' in markdown_file_path:
+        jinja.env.loader.searchpath.append(os.path.dirname(markdown_file_path))
```

Comment:
Hmm, is that a permanent change to the Jinja2 loader, or does that reset every time you view a file?  Might be worth looking if loading the same file with this multiple times leaks duplicate entries in that `searchpath` :).

I guess another thing to think about is whether we just want `settings.py` to do its own algorithm for detecting the path of the `doc.md` files from this dependency.